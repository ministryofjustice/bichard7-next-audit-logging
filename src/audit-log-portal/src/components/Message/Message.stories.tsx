import { action } from "@storybook/addon-actions"
import type { Meta } from "@storybook/react"
import { AuditLog, AuditLogStatus } from "shared-types"
import Message from "./Message"

export default {
  title: "components/Message",
  component: Message,
  argTypes: {
    reloadMessages: {
      action: "reload messages"
    }
  }
} as Meta

const createMessage = (status: string) => {
  const message = new AuditLog("LIBRA-EXISS-0001", new Date(), "XML")
  message.status = status

  return message
}

export const Processing = () => (
  <Message message={createMessage(AuditLogStatus.processing)} reloadMessages={action("reload messages")} />
)
export const Retrying = () => (
  <Message message={createMessage(AuditLogStatus.retrying)} reloadMessages={action("reload messages")} />
)
export const Completed = () => (
  <Message message={createMessage(AuditLogStatus.completed)} reloadMessages={action("reload messages")} />
)
export const Failed = () => (
  <Message message={createMessage(AuditLogStatus.error)} reloadMessages={action("reload messages")} />
)
