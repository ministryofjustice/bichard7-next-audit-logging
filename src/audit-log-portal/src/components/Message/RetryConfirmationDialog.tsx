import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@material-ui/core"
import InformationIcon from "icons/InformationIcon"
import styled from "styled-components"
import type AuditLog from "types/AuditLog"

interface Props {
  open: boolean
  selectedValue: AuditLog
  onClose: (shouldRetry: boolean) => void
}

const InfoText = styled.div`
  padding-left: 10px;
`
const InfoParagraph = styled.div`
  display: flex;
  align-items: center;
`

const RetryConfirmationDialog = ({ open, selectedValue, onClose }: Props) => (
  <Dialog aria-labelledby="retry-dialog-title" open={open}>
    <DialogTitle id="retry-dialog-title">Retry Message</DialogTitle>
    <DialogContent>
      <InfoParagraph>
        <InformationIcon />
        <InfoText>
          {`Are you sure you wish to retry "${selectedValue.externalCorrelationId}" from the last failure?`}
        </InfoText>
      </InfoParagraph>
    </DialogContent>
    <DialogActions>
      <Button onClick={() => onClose(true)} color="primary" variant="contained">
        {`Yes`}
      </Button>
      <Button onClick={() => onClose(false)}>{`No`}</Button>
    </DialogActions>
  </Dialog>
)

export default RetryConfirmationDialog
