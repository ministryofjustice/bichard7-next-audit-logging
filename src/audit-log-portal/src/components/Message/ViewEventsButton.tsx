import { Badge, IconButton, Tooltip } from "@material-ui/core"
import EventIcon from "icons/EventIcon"
import Link from "next/link"
import type AuditLog from "types/AuditLog"

interface Props {
  message: AuditLog
}

const ViewEventsButton = ({ message }: Props) => (
  <Badge badgeContent={(message.events || []).length} color="secondary">
    <Link href={`/messages/${message.messageId}`} passHref={true}>
      <Tooltip title="View events" aria-label="view">
        <IconButton>
          <EventIcon />
        </IconButton>
      </Tooltip>
    </Link>
  </Badge>
)

export default ViewEventsButton
