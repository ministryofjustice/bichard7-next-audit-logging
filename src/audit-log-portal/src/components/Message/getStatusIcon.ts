import ErrorIcon from "icons/ErrorIcon"
import ProcessingIcon from "icons/ProcessingIcon"
import SuccessIcon from "icons/SuccessIcon"
import type { ComponentType } from "react"
import AuditLogStatus from "types/AuditLogStatus"

export default function getStatusIcon(status: string): ComponentType {
  switch (status) {
    case AuditLogStatus.processing:
    case AuditLogStatus.retrying:
      return ProcessingIcon

    case AuditLogStatus.completed:
      return SuccessIcon

    default:
      return ErrorIcon
  }
}
