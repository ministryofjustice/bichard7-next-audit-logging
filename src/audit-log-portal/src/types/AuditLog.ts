import { v4 as uuid } from "uuid"
import type AuditLogEvent from "./AuditLogEvent"
import AuditLogStatus from "./AuditLogStatus"
import type AutomationReport from "./AutomationReport"
import type TopExceptionsReport from "./TopExceptionsReport"

export default class AuditLog {
  public readonly messageId: string

  public readonly receivedDate: string

  public forceOwner?: number

  public errorRecordArchivalDate?: string

  public isSanitised = 0

  public nextSanitiseCheck?: string

  public caseId: string

  public systemId: string

  public events: AuditLogEvent[] = []

  public automationReport?: AutomationReport = { events: [] }

  public topExceptionsReport?: TopExceptionsReport = { events: [] }

  public status = AuditLogStatus.processing

  public pncStatus?: string

  public triggerStatus?: string

  public lastEventType: string

  public createdBy: string

  public s3Path: string

  public externalId: string

  public stepExecutionId: string

  public retryCount?: number

  public readonly version = 0

  public expiryTime?: string

  public readonly externalCorrelationId: string

  public readonly messageHash: string

  constructor(externalCorrelationId: string, receivedDate: Date, messageHash: string, messageId?: string) {
    this.messageHash = messageHash
    this.externalCorrelationId = externalCorrelationId
    this.messageId = messageId ?? uuid()
    this.receivedDate = receivedDate.toISOString()
    this.nextSanitiseCheck = this.receivedDate
  }
}
