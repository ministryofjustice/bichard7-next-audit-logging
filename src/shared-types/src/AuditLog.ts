import { v4 as uuid } from "uuid"
import type AuditLogEvent from "./AuditLogEvent"
import AuditLogStatus from "./AuditLogStatus"
import type AutomationReport from "./AutomationReport"
import type TopExceptionsReport from "./TopExceptionsReport"

export default class AuditLog {
  public readonly messageId: string

  public readonly receivedDate: string

  public errorRecordArchivalDate?: string

  public isSanitised = 0

  public nextSanitiseCheck: string

  public caseId: string

  public systemId: string

  public events: AuditLogEvent[] = []

  public automationReport: AutomationReport = { events: [] }

  public topExceptionsReport: TopExceptionsReport = { events: [] }

  public status = AuditLogStatus.processing

  public lastEventType: string

  public createdBy: string

  public s3Path: string

  public externalId: string

  public stepExecutionId: string

  public retryCount?: number

  public readonly version = 0

  constructor(public readonly externalCorrelationId: string, receivedDate: Date, public readonly messageHash: string) {
    this.messageId = uuid()
    this.receivedDate = receivedDate.toISOString()
    this.nextSanitiseCheck = this.receivedDate
  }
}
