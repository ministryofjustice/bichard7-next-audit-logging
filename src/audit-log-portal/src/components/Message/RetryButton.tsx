import { useState } from "react"
import RetryIcon from "icons/RetryIcon"
import { Tooltip, IconButton } from '@material-core/ui'
import type { AuditLog } from "shared-types"
import If from "components/If"
import RetryConfirmationDialog from "./RetryConfirmationDialog"

interface Props {
  message: AuditLog
  show: boolean
  isRetrying: boolean
  onRetry: () => void
}

const RetryButton = ({ message, show, isRetrying, onRetry }: Props) => {
  const [isOpen, setIsOpen] = useState(false)

  return (
    <If condition={show}>
      <Tooltip title="Retry message" aria-label="retry">
        <IconButton onClick={() => setIsOpen(true)} disabled={isRetrying}>
          <RetryIcon spin={isRetrying} />
        </IconButton>
      </Tooltip>

      <RetryConfirmationDialog
        selectedValue={message}
        open={isOpen}
        onClose={(shouldRetry) => {
          setIsOpen(false)

          if (shouldRetry) {
            onRetry()
          }
        }}
      />
    </If>
  )
}

export default RetryButton
