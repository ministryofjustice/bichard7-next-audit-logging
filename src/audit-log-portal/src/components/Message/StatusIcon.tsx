import { Tooltip } from '@material-core/ui'
import type { AuditLog } from "shared-types"
import getStatusIcon from "./getStatusIcon"

interface Props {
  message: AuditLog
}

const StatusIcon = ({ message }: Props) => {
  const Icon = getStatusIcon(message.status)

  return (
    <Tooltip title={message.lastEventType || message.status} placement="top">
      <div>
        <Icon />
      </div>
    </Tooltip>
  )
}

export default StatusIcon
