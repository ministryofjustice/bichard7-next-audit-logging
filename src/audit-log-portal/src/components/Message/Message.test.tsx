import MockDate from "mockdate"
import "testing/mockUseFetch"
import { render } from "testing/render"
import AuditLog from "types/AuditLog"
import Message from "./Message"

test("matches snapshot", () => {
  MockDate.set(new Date("2021-05-24T10:20:30").getTime())

  const message = new AuditLog("ExternalCorrelationId", new Date("2021-05-01T10:25:53"), "XML")
  // @ts-ignore
  message.messageId = "test-uuid"

  const { container } = render(<Message message={message} reloadMessages={() => {}} />)

  expect(container).toMatchSnapshot()

  MockDate.reset()
})
