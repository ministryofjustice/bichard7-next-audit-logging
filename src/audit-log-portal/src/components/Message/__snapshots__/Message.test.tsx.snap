// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root Message__Container-sc-12ihk2z-0 jCoPiM MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root Message__InnerContainer-sc-12ihk2z-1 kFTIMa"
    >
      <div
        class="Message__Content-sc-12ihk2z-2 kIKsiJ"
      >
        <div
          class="Message__Block-sc-12ihk2z-3 Message__StatusBlock-sc-12ihk2z-4 dncGYJ ipLlpv"
        >
          <div
            class=""
            title="Processing"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
          <i>
            Processing
          </i>
        </div>
        <div
          class="Message__Block-sc-12ihk2z-3 dncGYJ"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          >
            ExternalCorrelationId
          </h6>
          <span
            class="MuiTypography-root Message__ReceivedDate-sc-12ihk2z-5 fqafoZ MuiTypography-caption MuiTypography-noWrap"
          >
            Received:
            <time
              aria-label="time"
            >
              01/05/2021 10:25:53
            </time>
          </span>
        </div>
        <h6
          class="MuiTypography-root Message__DaysAgo-sc-12ihk2z-6 ijFoKN MuiTypography-h6"
        >
          23 days old
        </h6>
        <div
          class="Message__Actions-sc-12ihk2z-7 dTgDfS"
        >
          <span
            class="MuiBadge-root"
          >
            <button
              aria-label="view"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              title="View events"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-calendar-day fa-w-14 "
                  data-icon="calendar-day"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style="width: 18px; height: 18px;"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
            >
              0
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
